pkgs/by-name/wa/wayneko/package.nix

```
@@ -0,0 +1,46 @@
+{ lib
+, stdenv
+, fetchFromSourcehut
+, pixman
+, pkg-config
+, wayland
+, wayland-scanner
+}:
+
+stdenv.mkDerivation {
+  pname = "wayneko";
+  version = "0-unstable-2024-03-29";
+
+  src = fetchFromSourcehut {
+    owner = "~leon_plickat";
+    repo = "wayneko";
+    rev = "c1919dc3a7e610d30e4c06efaa5af85941f27d86";
+    hash = "sha256-2cbEcDK6WZPe4HvY1pxmZVyDAj617VP1l0Gn7uSlNaE=";
+  };
+
+  postPatch = ''
+    substituteInPlace Makefile \
+      --replace-warn -Werror ""
```

```suggestion
      --replace-fail -Werror ""
```
we probably want to notice if that moves somewhere else or is entirely removed

---

Is there some guidance about the `--replace` variants somewhere? My reasoning for `warn` here is that even if this replace fails it isn't going to affect the package in any way. At most it's going to fail to build. Using `fail` here can prevent automatic updates for no actual benefit.

On the other hand, if I was for example replacing a hardcoded path of a runtime dependency I would use `--replace-fail`, because a failed replace would silently produce a broken package.

---

Ping

---

Ping

---

IMO `--replace-fail` should be used in all situations when the code to be patched is definite

---

What do you mean by definite/indefinite? Aren't all patches "definite"?

---

I mean "we know exactly what we are patching". If we are patching a set of unspecified files with common characteristics, it would make sense to use warn or even success. But if the patch fails for a single, specific file after it is updated, the least bad result is we get a dead code. But it is likely that we already need additional processing. Here, failure makes it easier for us to see the problem in time.